sqa/fuel/writer/parquet.py

```diff
@@ -0,0 +1,584 @@
+            self.coin_predicate.append(None)
+            self.coin_predicate_data.append(None)
+
+        if input['type'] == 'InputContract':
+            self.contract_utxo_id.append(input['utxoId'])
+            self.contract_balance_root.append(input['balanceRoot'])
+            self.contract_state_root.append(input['stateRoot'])
+            self.contract_tx_pointer.append(input['txPointer'])
+            self.contract_contract.append(input['contract'])
+        else:
+            self.contract_utxo_id.append(None)
+            self.contract_balance_root.append(None)
+            self.contract_state_root.append(None)
+            self.contract_tx_pointer.append(None)
+            self.contract_contract.append(None)
+
+        if input['type'] == 'InputMessage':
+            self.message_sender.append(input['sender'])
+            self.message_recipient.append(input['recipient'])
+            self.message_amount.append(int(input['amount']))
+            self.message_nonce.append(input['nonce'])
+            self.message_witness_index.append(input['witnessIndex'])
+            self.message_predicate_gas_used.append(int(input['predicateGasUsed']))
+            self.message_data.append(input['data'])
+            self.message_predicate.append(input['predicate'])
+            self.message_predicate_data.append(input['predicateData'])
+        else:
+            self.message_sender.append(None)
+            self.message_recipient.append(None)
+            self.message_amount.append(None)
+            self.message_nonce.append(None)
+            self.message_witness_index.append(None)
+            self.message_predicate_gas_used.append(None)
+            self.message_data.append(None)
+            self.message_predicate.append(None)
+            self.message_predicate_data.append(None)
+
+
+class OutputTable(TableBuilder):
+    def __init__(self):
+        self.block_number = Column(pyarrow.int32())
+        self.transaction_index = Column(pyarrow.int32())
+        self.index = Column(pyarrow.int32())
+        self.type = Column(pyarrow.string())
+        # coin output
+        self.coin_to = Column(pyarrow.string())
+        self.coin_amount = Column(pyarrow.uint64())
+        self.coin_asset_id = Column(pyarrow.string())
+        # contract output
+        self.contract_input_index = Column(pyarrow.int32())
+        self.contract_balance_root = Column(pyarrow.string())
+        self.contract_state_root = Column(pyarrow.string())
+        # change output
+        self.change_to = Column(pyarrow.string())
+        self.change_amount = Column(pyarrow.uint64())
+        self.change_asset_id = Column(pyarrow.string())
+        # variable output
+        self.variable_to = Column(pyarrow.string())
+        self.variable_amount = Column(pyarrow.uint64())
+        self.variable_asset_id = Column(pyarrow.string())
+        # contract created
+        self.contract_created_contract_id = Column(pyarrow.string())
```

Comment:
does it make sense to flatten `contract` fields and allow users to load created contracts without their bytecode? i guess we don't need to do it and small traffic overhead will not be a problem